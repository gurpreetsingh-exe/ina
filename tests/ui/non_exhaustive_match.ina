// COMPILE_FAIL

type T = A | B
using Option::*;
type Option[T] = Some(T) | None
type T2 = A(T, T)

fn main() {
    let x = T::A;
    // error
    match x {
        T::A => {},
    }
    // error
    match Some(20) {
        None => {}
    }
    // ok
    match T::A {
        _ => {}
    }
    // ok
    match T::B {
        _b => {}
    }
    // error
    match Some(T::A) {
        Some(T::A) => {},
        None => {},
    }
    // ok
    match T2::A(T::A, T::A) {
        T2::A(T::A, _) => {},
        T2::A(_, T::A) => {},
        T2::A(T::B, T::B) => {},
    }
    // error
    match None {
        Some(T2::A(T::A, _)) => {},
        Some(T2::A(_, T::B)) => {},
    }
}
