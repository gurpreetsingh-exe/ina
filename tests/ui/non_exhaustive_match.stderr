error: non-exhaustive patterns
    ╭─[tests/ui/non_exhaustive_match.ina:11:11]
    │
 11 │     match x {
    │           ┬
    │           ╰─ pattern `B` not covered
    ∙
────╯
error: non-exhaustive patterns
    ╭─[tests/ui/non_exhaustive_match.ina:15:11]
    │
 15 │     match Some(20) {
    │           ────┬───
    │               ╰───── pattern `Some(_)` not covered
    ∙
────╯
error: non-exhaustive patterns
    ╭─[tests/ui/non_exhaustive_match.ina:27:11]
    │
 27 │     match Some(T::A) {
    │           ─────┬────
    │                ╰────── pattern `Some(B)` not covered
    ∙
────╯
error: non-exhaustive patterns
    ╭─[tests/ui/non_exhaustive_match.ina:38:11]
    │
 38 │     match None {
    │           ──┬─
    │             ╰─── patterns `None` and `Some(A(B, A))` not covered
    ∙
────╯
